sdk/nodejs/apigateway/experimental/api.ts

```diff
@@ -292,46 +315,72 @@ interface ApigatewayIntegration {
     connectionId?: pulumi.Output<string>;
 }
 
-function createSwaggerSpec(
-        name: string,
-        opts: pulumi.ResourceOptions,
-        routes?: Route[],
-        staticRoutes?: StaticRoute[],
-        proxyRoutes?: ProxyRoute[]): SwaggerSpec {
-
+function createSwaggerSpec(name: string, opts: pulumi.ResourceOptions, routes: Routes): SwaggerSpec {
     // Set up the initial swagger spec.
     const swagger: SwaggerSpec = {
         swagger: "2.0",
         info: { title: name, version: "1.0" },
         paths: {},
         "x-amazon-apigateway-binary-media-types": ["*/*"],
+        // Map paths the user doesn't have access to as 404.
+        // http://docs.aws.amazon.com/apigateway/latest/developerguide/supported-gateway-response-types.html
+        "x-amazon-apigateway-gateway-responses": {
+            "MISSING_AUTHENTICATION_TOKEN": {
+                "statusCode": 404,
+                "responseTemplates": {
+                    "application/json": "{\"message\": \"404 Not found\" }",
+                },
+            },
+            "ACCESS_DENIED": {
+                "statusCode": 404,
+                "responseTemplates": {
+                    "application/json": "{\"message\": \"404 Not found\" }",
+                },
+            },
+        },
```

Comment:
this just moved since it was conceptually simpler to just supply this at creation time, rather than adding it in after the fact.